safenode/src/storage/spends.rs

```diff
@@ -79,7 +72,36 @@ impl SpendStorage {
     /// If a double spend attempt is detected, a `DoubleSpendAttempt` error
     /// will be returned including all the `SignedSpends`, for
     /// broadcasting to the other nodes.
-    pub(super) async fn try_add(&self, signed_spend: &SignedSpend) -> Result<()> {
+    /// NOTE: The &mut self signature is necessary to prevent race conditions
```

Comment:
you mean the step of `self.validate(signed_spend).await?;` ?

also, seems missing \` \` around `&mut self` ?

Reply:
Clarified comments there, and added the ... what ever they are called :)